diff --git a/dist/index.js b/dist/index.js
index 5ed9b0106cbe206b1d460c234be24869708f6973..aa6e29c7bfd06868f90fc21f86356045d483a7fb 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -256,7 +256,7 @@ function convertToOpenAICompatibleChatMessages(prompt) {
         }
         messages.push({
           role: "assistant",
-          content: toolCalls.length > 0 ? text || null : text,
+          content: toolCalls.length > 0 ? text || '' : text,
           ...reasoning.length > 0 ? { reasoning_content: reasoning } : {},
           tool_calls: toolCalls.length > 0 ? toolCalls : void 0,
           ...metadata
@@ -283,6 +283,9 @@ function convertToOpenAICompatibleChatMessages(prompt) {
             case "error-json":
               contentValue = JSON.stringify(output.value);
               break;
+            default:
+              contentValue = typeof output.value === 'string' ? output.value : JSON.stringify(output.value) ?? '';
+              break;
           }
           const toolResponseMetadata = getOpenAIMetadata(toolResponse);
           messages.push({
diff --git a/dist/index.mjs b/dist/index.mjs
index cecabb98227512c8bdbad38861e4bc8418641b2f..a8777b80914f311f4a6a8e5cc044fd7aff6811ee 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -241,7 +241,7 @@ function convertToOpenAICompatibleChatMessages(prompt) {
         }
         messages.push({
           role: "assistant",
-          content: toolCalls.length > 0 ? text || null : text,
+          content: toolCalls.length > 0 ? text || '' : text,
           ...reasoning.length > 0 ? { reasoning_content: reasoning } : {},
           tool_calls: toolCalls.length > 0 ? toolCalls : void 0,
           ...metadata
@@ -268,6 +268,9 @@ function convertToOpenAICompatibleChatMessages(prompt) {
             case "error-json":
               contentValue = JSON.stringify(output.value);
               break;
+            default:
+              contentValue = typeof output.value === 'string' ? output.value : JSON.stringify(output.value) ?? '';
+              break;
           }
           const toolResponseMetadata = getOpenAIMetadata(toolResponse);
           messages.push({
diff --git a/dist/internal/index.js b/dist/internal/index.js
index c3a756439f982694f3d22f391b76dafacc13ed41..f2c198ebd2e46413b70add089c2cc36198e44acb 100644
--- a/dist/internal/index.js
+++ b/dist/internal/index.js
@@ -181,7 +181,7 @@ function convertToOpenAICompatibleChatMessages(prompt) {
         }
         messages.push({
           role: "assistant",
-          content: toolCalls.length > 0 ? text || null : text,
+          content: toolCalls.length > 0 ? text || '' : text,
           ...reasoning.length > 0 ? { reasoning_content: reasoning } : {},
           tool_calls: toolCalls.length > 0 ? toolCalls : void 0,
           ...metadata
@@ -208,6 +208,9 @@ function convertToOpenAICompatibleChatMessages(prompt) {
             case "error-json":
               contentValue = JSON.stringify(output.value);
               break;
+            default:
+              contentValue = typeof output.value === 'string' ? output.value : JSON.stringify(output.value) ?? '';
+              break;
           }
           const toolResponseMetadata = getOpenAIMetadata(toolResponse);
           messages.push({
diff --git a/dist/internal/index.mjs b/dist/internal/index.mjs
index 5163d3227c547bcc4bcc91f05a4ffa459c09b00e..4d63371d717d58fa43a1e9d5fdca8e6dc68970b3 100644
--- a/dist/internal/index.mjs
+++ b/dist/internal/index.mjs
@@ -156,7 +156,7 @@ function convertToOpenAICompatibleChatMessages(prompt) {
         }
         messages.push({
           role: "assistant",
-          content: toolCalls.length > 0 ? text || null : text,
+          content: toolCalls.length > 0 ? text || '' : text,
           ...reasoning.length > 0 ? { reasoning_content: reasoning } : {},
           tool_calls: toolCalls.length > 0 ? toolCalls : void 0,
           ...metadata
@@ -183,6 +183,9 @@ function convertToOpenAICompatibleChatMessages(prompt) {
             case "error-json":
               contentValue = JSON.stringify(output.value);
               break;
+            default:
+              contentValue = typeof output.value === 'string' ? output.value : JSON.stringify(output.value) ?? '';
+              break;
           }
           const toolResponseMetadata = getOpenAIMetadata(toolResponse);
           messages.push({
diff --git a/src/chat/convert-to-openai-compatible-chat-messages.ts b/src/chat/convert-to-openai-compatible-chat-messages.ts
index 638ce2b0dd29d1f294a6be2b1974990c50b847c0..8bc7d1323ff183eb9a72cdbde3432ee821f6e380 100644
--- a/src/chat/convert-to-openai-compatible-chat-messages.ts
+++ b/src/chat/convert-to-openai-compatible-chat-messages.ts
@@ -202,7 +202,7 @@ export function convertToOpenAICompatibleChatMessages(
 
         messages.push({
           role: 'assistant',
-          content: toolCalls.length > 0 ? text || null : text,
+          content: toolCalls.length > 0 ? text || '' : text,
           ...(reasoning.length > 0 ? { reasoning_content: reasoning } : {}),
           tool_calls: toolCalls.length > 0 ? toolCalls : undefined,
           ...metadata,
@@ -233,6 +233,9 @@ export function convertToOpenAICompatibleChatMessages(
             case 'error-json':
               contentValue = JSON.stringify(output.value);
               break;
+            default:
+              contentValue = typeof output.value === 'string' ? output.value : JSON.stringify(output.value) ?? '';
+              break;
           }
 
           const toolResponseMetadata = getOpenAIMetadata(toolResponse);
